pkg/trace/api/api.go

```diff
@@ -442,3 +444,11 @@ func tracesFromSpans(spans []pb.Span) pb.Traces {
 
 	return traces
 }
+
+// logWriter is an error logging function which can be written to.
+type logWriter func(v ...interface{}) error
+
+// Write implements io.Writer
+func (lw logWriter) Write(p []byte) (n int, err error) {
+	return len(p), lw(string(p))
+}
```

Comment:
This is a bit weird x)

Reply:
Which part do you mean? To me it's weird that the first return value might not be 100% accurate but I can't think of a better solution. I'm open to suggestions!

Reply:
I mean the whole thing about having a type that is aliased to a function and calling `lw(...)` directly from a method on that type/function. Then the part where we do `logWriter(log.Error)` which is actually doing a type conversion not an invocation.

I see how/why it works after looking at it carefully but initially I got a bit confused as to what was going on.

Reply:
Initially I had a struct but then I thought of [`http.HandlerFunc`](http://golang.org/pkg/net/http#HandlerFunc) which is somewhat similar and went with that. Maybe `funcWriter` would've been better but 🤷‍♂️

Reply:
I had to rebase on a new `master` and made this a bit better (I think) both in terms of naming and error returns. Can you PTAL? Your review became stale as a result of the rebase.